resolver/resolver.go

```diff
@@ -51,8 +47,25 @@ func New(version string, config records.Config) *Resolver {
 		rs:      &records.RecordGenerator{},
 		rng:     rand.New(rand.NewSource(time.Now().UnixNano())),
 	}
-	r.extResolver = r.defaultExtResolver
+
+	timeout := 5 * time.Second
+	if config.Timeout != 0 {
+		timeout = time.Duration(config.Timeout) * time.Second
+	}
+
+	client := &dns.Client{
+		DialTimeout:  timeout,
+		ReadTimeout:  timeout,
+		WriteTimeout: timeout,
+	}
+
+	r.extResolver = exchanger.Decorate(client,
```

Comment:
I attempted that before and realised that the `Exchanger` interface doesn't play well with that idea since it accepts the address to direct a DNS query to in its `Exchange` method.

+1 on the second suggestion though, which is still doable .